server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -503,6 +509,33 @@ public boolean shouldSeedRemoteStore() {
         return shardMigrationState == REMOTE_MIGRATING_UNSEEDED;
     }
 
+    /**
+     * To be delegated to {@link ReplicationTracker} so that relevant remote store based
+     * operations can be ignored during engine migration
+     * <p>
+     * Has explicit null checks to ensure that the {@link ReplicationTracker#invariant()}
+     * checks does not fail during a cluster manager state update when the latest replication group
+     * calculation is not yet done and the cached replication group details are available
+     */
+    public Function<String, Boolean> isShardOnRemoteEnabledNode = (shardId) -> {
+        DiscoveryNode discoveryNode = this.discoveryNodes.get(shardId);
+        if (discoveryNode != null) {
+            logger.trace(
+                "ShardID {} is assigned to Node {} which has remote_enabled as {}",
+                shardId,
+                discoveryNode,
+                discoveryNode.isRemoteStoreNode()
+            );
+            return discoveryNode.isRemoteStoreNode();
+        }
+        return false;
+    };
+
+    // Only to be used for Unit Tests
+    public void setDiscoveryNodes(DiscoveryNodes discoveryNodes) {
```

Comment:
This is a bad practise to expose setters. Why can't ctor ingestion not work?

Reply:
Ack. This was only used for certain UTs. Will remove this and change the UTs accordingly.